pkg/server/backend_manager.go

```diff
@@ -204,7 +214,6 @@ func (s *DefaultBackendStorage) AddBackend(identifier string, idType pkgagent.Id
 		return addedBackend
 	}
 	s.backends[identifier] = []*backend{addedBackend}
-	metrics.Metrics.SetBackendCount(len(s.backends))
```

Comment:
Thanks for your contribution, @YRXING .
We are complicating the logic of the backend count metrics. we don't have to rewrite `AddBackend` for each manager to collect backend count metrics. code maintenance and reusability should be taken into consideration.

How about we keep `SetBackendCount` here and add the manager name, idtype into it. manager name, can be defined as member of backendstore.

Reply:
Does the DefaultBackendStorage need record different idTypes count metric or just record the total backend count?

Reply:
Following your thought already in this PR, record different backend counts from different idTypes and backend managers. `SetBackendCount` will finally look like this way after other changes are also included.
```suggestion
	metrics.Metrics.SetBackendCount(s.backendType, idType, len(s.backends))
````
cc @cheftako and @jkh52 who will give the final say to this PR.

Reply:
we can only set the total number of backends stored in this manager through `SetBackendCount(s.backendType, idType, len(s.backends))` method. Because the storage can not distinguish different types of backends. (unless we change the way it stores the backends)

```go
type DefaultBackendStorage struct {
    //...
    backends map[string][]*backend
    //...
}
```

If we want to record diffrent backends, we can use `BackendCountDec` and `BackendCountInc` method which I have defined in `metrics.go` file. It's just that the caller becomes DefaultBackendStorage.

Is that Ok?